--- a/arch/arm/boot/dts/sun8i-h2-plus-orangepi-zero.dts
+++ b/arch/arm/boot/dts/sun8i-h2-plus-orangepi-zero.dts
@@ -59,6 +59,8 @@
 		/* ethernet0 is the H3 emac, defined in sun8i-h3.dtsi */
 		ethernet0 = &emac;
 		ethernet1 = &xr819;
+                spi1 = &spi1;
+                i2c1 = &i2c1;
 	};
 
 	chosen {
@@ -235,3 +237,15 @@
     };
 };
 
+&spi1 {
+	status = "okay";
+        spidev@0x00 {
+                compatible = "spidev";
+                spi-max-frequency = <5000000>;
+                reg = <0>;
+       };
+};
+
+&i2c1 {
+	status = "okay";
+};
